clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -520,10 +520,8 @@ private void resetOffsetPosition(TopicPartition tp) {
     }
 
     private Long getEndOffset(List<Long> offsets) {
-        if (offsets == null || offsets.isEmpty()) {
-            return null;
-        }
-        return offsets.size() > 1 ? offsets.remove(0) : offsets.get(0);
+        if (offsets == null || offsets.isEmpty()) return null;
+        else return offsets.get(offsets.size() - 1);
```

Comment:
The code seems intentionally written to store a sequence of end offsets so that each call advances the end offset. If we do not need that capability, then perhaps we can change the type of `endOffsets` and simplify the usage.

cc @bbejeck Looks like you added this logic. Do you know if we rely on it?

Reply:
I agree with @hachikuji 's observation. This test broke when https://github.com/apache/kafka/pull/8220 retired the `innerUpdateEndOffsets` and replaced the two public methods `addEndOffsets` and `updateEndOffsets` that were both using `innerUpdateEndOffsets` with a single `updateEndOffsets`. However the version that was kept was actually what the unused `addEndOffsets` was doing (appending to the end offsets) and not what `updateEndOffsets` described as an overwrite of endOffsets. 

Thus, probably the issue should be fixed in `updateEndOffsets` to keep meaning overwrite. Unless we find a usage for `addEndOffsets`, in which case we might some more adjustments too. Of course, if we keep only `updateEndOffsets` that overwrite the value of the map, we might as well simplify `getEndOffset`. Haven't tracked when and why `addEndOffsets` was stopped being used.

Reply:
@hachikuji @kkonstantine thanks for reviews.

Will address your comment :)